### DISCUSSION TITLE
20230831 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by cgbosse** [nautilus-cyberneering/.github#56] [Meetings]
Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - x
- JCelano(@josecelano) - x 
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by da2ce7** [comment]
My work has again been focused on the Container Overhaul for the Tracker: https://github.com/torrust/torrust-tracker/pull/363

**Comment by josecelano** [comment]
I've been off most of the days since our last meeting. I've caught up with all the issues/PR reviews/comments and continued working on the academic torrent importer. I found another "invalid" torrent and opened the issue:

https://github.com/torrust/torrust-index-backend/issues/266

I've also opened an issue to review all the dirs and files the app needs to work:

https://github.com/torrust/torrust-tracker/issues/392

because I think we could/must simplify it.

**Comment by mario-nt** [comment]
### Torrust Front End

- Opened https://github.com/torrust/torrust-index-frontend/issues/247
- Pushed https://github.com/torrust/torrust-index-frontend/pull/246
 
### Torrust Back End

- Opened https://github.com/torrust/torrust-index-backend/issues/264
- Pushed https://github.com/torrust/torrust-index-backend/pull/265

### Torrust Index Types Lib

- Opened https://github.com/torrust/torrust-index-types-lib/issues/9
- Pushed https://github.com/torrust/torrust-index-types-lib/pull/10

### Torrust Website

- Opened https://github.com/torrust/torrust-website/issues/81
- Pushed  https://github.com/torrust/torrust-website/pull/82

### Git/Github

- Research on different git topics ( merge vs rebase, how Github PRs work, pushing branches to different repos, etc)
- Setup of forked repos

### Training

- Research on several Nuxt/Vue topics (Composition vs options APIs, router, routes, watchers, components, setup scripts, etc)

**Reply by cgbosse** [reply]
@mario-nt  some food for thought:
When you are learning about Torrust, Rust and other things, imagine that possibly in the future somebody else in the team could be interested in learning it. How can you make your experience of use for newcomers? How could you make it easier for the new comers to learn etc.?

**Reply by mario-nt** [reply]
I would suggest start working on small issues or start making small apps to get used to the language. 

After doing that for a few days, I would take a small course or watch a few introductory videos. 

I could use my experience to give advice on what issues are good for beginners, and they could ask me for help.

I would also recommend to go and look at the official documentation and to focus on understanding the problem before trying to solve it.

**Comment by grmbyrn** [comment]
## Git Secure Guide
* Learned and applied routing in Svelte to the project through appropriate packages.
* Researched how to bring in the markdown from each article to appear on individual pages.
* Learned how to store information in Svelte Stores and implemented it to bring in and display in appropriate HTML elements based on Svelte logic.

## SvelteKit
* Decided after struggling to implement markdown in project either through packages or through Stores that SvelteKit would be the best option for this as it has built-in markdown support, as well as a range of other features such as superior routing, SSR and data-fetching which make it better suited for the project.
* Found resources to learn more about SK. Primarily learned about routing and layouts with the intention of continuing the course, building some projects, and then implement it in Git Secure Guide.

## Medium
* Researched and wrote blog posts on Svelte Stores and the advantages of using SvelteKit over Svelte.

**Reply by cgbosse** [reply]
@grmbyrn some food for thought:
When you are learning about Svelte Kit, imagine that possibly in the future somebody else in the team could be interested in learning it.
How can you make your experience  of use for newcomers? How could you make it easier for the new comers to learn etc.?

**Reply by grmbyrn** [reply]
For me learning SK, it helped that I had learned React as it gave me something to compare with, so I would ask them if they have experience with another JS framework. If yes, I would advise them to find out the differences between the framework they have learned and SK. This could be through finding a project which uses SK, a SK course, reading the doc, following a tutorial, etc, depending on their learning style.

From there, I would recommend building as soon as possible in order to get the practice in so they don't forget what they've learned. A good idea is to take something they've already built in another framework and see if they can rebuild it in SK.

If this is their first framework, I would make clear the advantages of using a framework and how they work as opposed to not using one. The main advantage I'd highlight would be the ability to break your application into smaller, reusable components so that you can organise and maintain your code better. As I have been in their position of not knowing anything about these things, I could use my experience to reassure them that with time, patience and persistence, they will be able to understand and use SK.
